libraries/eosiolib/contracts/include/eosio/action.hpp

```diff
@@ -225,6 +230,35 @@ namespace eosio
     */
    inline bool is_account(name n) { return internal_use_do_not_use::is_account(n.value); }
 
+   struct code_hash
+   {
+      varuint32 struct_version = {};
+      uint64_t code_sequence = {};
+      checksum256 code_hash = {};
```

Comment:
This compiles?

Reply:
yes. c++20 added additional rules making non-movable, non-copyable objects sometimes returnable, including from constructors.

Reply:
Oh, the same name... I'm not sure why it compiles.

Reply:
Apparently its okay as long as you have no user-declared constructors.  Probably for C compatibility.